public/app/plugins/panel/geomap/layers/data/circlesOverlay.ts

```diff
@@ -48,32 +48,30 @@ export const circlesLayer: MapLayerRegistryItem<CircleConfig> = {
       init: () => vectorLayer,
       update: (data: PanelData) => {
         const features: Feature[] = [];
-        const dataFrame = data.series[0];
-        // Get data values
-        const points = dataFrameToPoints(dataFrame, config);
+        const frame = data.series[0];
 
-        // TODO: Find better way to find value field
-        const values = dataFrame.fields.find(a => a.name === config.fieldMapping.metricField);
+        // Get data values
+        const points = dataFrameToPoints(frame, config);
+        const values = frame.fields.find(field => field.name === config.fieldMapping.metricField)!;
```

Comment:
maybe call it `field` so that you don't have `values.values` below :)
```suggestion
        const field = frame.fields.find(field => field.name === config.fieldMapping.metricField)!;
```

Reply:
rather than adding the "!" to assert you foud a field... maybe check if the result exists and return early if nothing was matched

Reply:
makes sense, thank you!